src/formatDate.js

```diff
@@ -50,7 +51,40 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldDateArr = date.split(fromFormat[3]);
+  const oldFormat = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    oldFormat[fromFormat[i]] = oldDateArr[i];
+  }
+
+  const newFormatArr = [];
+  const toFormat1 = toFormat.map(x => x[0]);
+
+  for (const key in oldFormat) {
+    const newIndex = toFormat1.indexOf(key[0]);
+
+    newFormatArr[newIndex] = oldFormat[key];
+  }
+
+  const newFormat = {};
+
+  for (let i = 0; i < newFormatArr.length; i++) {
+    newFormat[toFormat[i]] = newFormatArr[i];
+  }
+
+  switch (true) {
+    case fromFormat.join('').length > toFormat.join('').length:
+      newFormat.YY = (oldFormat.YYYY).slice(2);
+      break;
+    case fromFormat.join('').length < toFormat.join('').length:
+      newFormat.YYYY
+      = oldFormat.YY >= 30 ? '19' + oldFormat.YY : '20' + oldFormat.YY;
+  }
```

Comment:
It is better to use [ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) if you want to process two conditions